cpp/unreal_plugins/SimulationController/Source/SimulationController/CameraSensor.cpp

```diff
@@ -386,6 +386,31 @@ void CameraSensor::initializeSceneCaptureComponentNonFinalColor(USceneCaptureCom
     ASSERT(material);
     scene_capture_component->PostProcessSettings.AddBlendable(UMaterialInstanceDynamic::Create(material, scene_capture_component), 1.0f);
     scene_capture_component->ShowFlags.SetPostProcessMaterial(true);
+
```

Comment:
Move this block of code directly into the constructor. You're not really initializing a `SceneCaptureComponent2D` in this block of code, so it's misleading to put it in that function.
 
```
        if (render_pass_name == "final_color") {
            initializeSceneCaptureComponentFinalColor(render_pass.scene_capture_component_);
        } else {
            initializeSceneCaptureComponentNonFinalColor(render_pass.scene_capture_component_, render_pass_name);
        }

        // PUT YOUR BLOCK OF CODE HERE

        // create shared_memory_object
        if (Config::get<bool>("SIMULATION_CONTROLLER.CAMERA_SENSOR.USE_SHARED_MEMORY")) {
            render_pass.shared_memory_num_bytes_ =
                height * width * RENDER_PASS_NUM_CHANNELS.at(render_pass_name) * RENDER_PASS_NUM_BYTES_PER_CHANNEL.at(render_pass_name);
            render_pass.shared_memory_name_ = "camera." + render_pass_name;

            #if BOOST_OS_WINDOWS
                render_pass.shared_memory_id_ = render_pass.shared_memory_name_; // don't use leading slash on Windows
                boost::interprocess::windows_shared_memory windows_shared_memory(
                    boost::interprocess::create_only,
                    render_pass.shared_memory_id_.c_str(),
                    boost::interprocess::read_write,
                    render_pass.shared_memory_num_bytes_);
                render_pass.shared_memory_mapped_region_ = boost::interprocess::mapped_region(windows_shared_memory, boost::interprocess::read_write);
            #elif BOOST_OS_MACOS || BOOST_OS_UNIX
                render_pass.shared_memory_id_ = "/" + render_pass.shared_memory_name_; // use leading slash on macOS and Linux
                boost::interprocess::shared_memory_object::remove(render_pass.shared_memory_id_.c_str());
                boost::interprocess::shared_memory_object shared_memory_object(
                    boost::interprocess::create_only,
                    render_pass.shared_memory_id_.c_str(),
                    boost::interprocess::read_write);
                shared_memory_object.truncate(render_pass.shared_memory_num_bytes_);
                render_pass.shared_memory_mapped_region_ = boost::interprocess::mapped_region(shared_memory_object, boost::interprocess::read_write);
            #else
                #error
            #endif
        }
```

Reply:
Right, makes sense. Changed.